File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/Issue458.java
package org.opencds.cqf.cql.engine.execution;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.opencds.cqf.cql.engine.runtime.Interval;

class Issue458 extends CqlTestBase {

    @Test
    void interval() {
        var results = engine.evaluate(toElmIdentifier("Issue458"));
        var value = results.forExpression("Closed-Open Interval").value();
        Interval interval = (Interval) value;
        assertEquals("Interval[3, 5)", interval.toString());
    }
}

File: Src/java/engine/src/test/kotlin/org/opencds/cqf/cql/engine/execution/Issue458.kt
package org.opencds.cqf.cql.engine.execution

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.opencds.cqf.cql.engine.runtime.Interval

internal class Issue458 : CqlTestBase() {
    @Test
    fun interval() {
        val results = engine.evaluate(toElmIdentifier("Issue458"))
        val value = results.forExpression("Closed-Open Interval")!!.value()
        val interval = value as Interval
        Assertions.assertEquals("Interval[3, 5)", interval.toString())
    }
}
